app/controllers/favourites_controller.rb

```diff
@@ -0,0 +1,18 @@
+class FavouritesController < ApplicationController
+  def change_fav
+    @did = params[:design_id]
+    if current_user.activation
+      favourite = current_user.favourited(@did)
+      if favourite
+        favourite.destroy
+      else
+        fav = Favourite.new(user_id: current_user.id, design_id: params[:design_id] )
```

Comment:
use #create instead of #new and #save as you are not doing anything with that object. Create it directly.